### ISSUE TITLE
The command execution not working

### ISSUE THREAD

**Issue by mpgn** [mpgn/CrackMapExec#58]
### Discussed in https://github.com/mpgn/CrackMapExec/discussions/57

<div type='discussions-op-text'>

<sup>Originally posted by **0xRyuzak1** July 24, 2023</sup>
it just keep hanging like this without any output 

![image](https://github.com/mpgn/CrackMapExec/assets/40356257/c2713c63-680e-45d6-b925-dda19435ac01)
</div>

**Comment by mpgn** [comment]
Hello @0xRyuzak1 di you try with flag --local-auth ?
Which version are you using ?

**Comment by 0xRyuzak1** [comment]
Hi @mpgn Yes i tried it with --local-auth flag and it the same issue iam using the latest version 6.0.1

**Comment by mpgn** [comment]
any av on the target machine ?

**Comment by 0xRyuzak1** [comment]
@mpgn No man there is no any AV on the target machine actually i think now after a lot of troubleshooting 
I found the cause of the issue it is on the crackmapexec tool itself

**Comment by mpgn** [comment]
highlight me then

**Comment by 0xRyuzak1** [comment]
Sure I am happy to help cuz this tool is my favorite tool ❤️

Actually the cuz of this issue is on this file `/home/kali/.local/pipx/venvs/crackmapexec/lib/python3.11/site-packages/impacket/dcerpc/v5/transport.py` on line `line 351` the tool got hanged in there because this part of code `self.__socket.settimeout(self.get_connect_timeout())` has a not correct timeout setting which is appears to be `None` from the `Passed args: Namespace` and since when used the `-x` to execute command the tool trying wmiexec first as default and if this not allowed the tool hanged in there very long time before it goes through the other methods like smbexec  

so to sum up the timeout in the wmiexec here is not configured correct so when user try to execute command without specify the exec-method if the wmiexec not allowed for any kind of reasons the tools will changed for very long time before tryin the other methods for executions

**Comment by mpgn** [comment]
I will check, thanks but it is strange you cannot exec with wmiexec if you are admin.

Same behavior with wmiexec.py ?

**Comment by 0xRyuzak1** [comment]
actually got the same with wmiexec.py at the same point it's hangs for long time then give me a time out it is very wired for me because i tested it on many machines and got the same results . But for crackmapexec it's as i said this what make the tool hanged for very long time before trying the other methods of executions .

![image](https://github.com/mpgn/CrackMapExec/assets/40356257/4beef766-24b9-4e47-b74f-129ddc46204d)

**Comment by mpgn** [comment]
Yes, I agree this is an issue, i will fix it thanks for the report !

**Comment by 0xRyuzak1** [comment]
Is this can make me a contributor on crackmapexec by any way 😅

**Comment by termanix** [comment]
I got the same thing on my labs.  Can not connect with smbexec or wmiexec. I've never encountered anything like this before, it just happened. Then i fix this temporarily turn off the windows firewall and worked again. However, in real life tests, Windows firewall is open and I can run commands, I only encountered this problem in a lab environment.

**Comment by 0xRyuzak1** [comment]
@mpgn Yes agreed but maybe a check from crackmapexec if no response came from wmiexec for certain period of time to skip it and go for other exec methods .

**Comment by XiaoliChan** [comment]
This is caused by the firewall, WMI needs to setup RPC transport after getting string-binding from port 135, so we can add a `try` to setup a new RPC transport, then it can avoid hard-coding timeout "300"

I will try to debug it.

https://github.com/fortra/impacket/blob/master/impacket/dcerpc/v5/dcomrt.py#L1288C39-L1288C39

**Comment by XiaoliChan** [comment]
@0xRyuzak1 Can you try this PR?
